kernel/packages/unity-interface/realmsForRenderer.ts

```diff
@@ -0,0 +1,60 @@
+import { StoreContainer } from '../shared/store/rootTypes'
+import { getExploreRealmsService, getRealm } from '../shared/dao/selectors'
+import { RealmsInfoForRenderer } from '../shared/types'
+import { observeRealmChange } from 'shared/dao'
+import { Realm } from 'shared/dao/types'
+import { unityInterface } from './UnityInterface'
+
+const REPORT_INTERVAL = 2 * 60 * 1000
+
+declare const globalThis: StoreContainer
+
+let isReporting = false
+
+export function startRealmsReportToRenderer() {
+  if (!isReporting) {
+    isReporting = true
+
+    const realm = getRealm(globalThis.globalStore.getState())
+    if (realm) {
+      reportToRenderer({ current: convertRealmType(realm) })
+    }
+
+    observeRealmChange(globalThis.globalStore, (previous, current) => {
+      reportToRenderer({ current: convertRealmType(current) })
+    })
+
+    fetchAndReportRealmsInfo()
+
+    reportInterval()
+  }
+}
+
+function reportInterval() {
+  setInterval(() => {
```

Comment:
I'd suggest replacing the `setInterval` for a `setTimeout`, because then you can do this:
```typescript
function reportInterval() {
  setTimeout(async () => {
    await fetchAndReportRealmsInfo()
    reportInterval()
  }, REPORT_INTERVAL)
}
```
Ensuring that even if the fetch takes time, there is no chance that two updates are executed at the same time

(you'd need to make `fetchAndReportRealmsInfo` async for this to work, but that should be relatively easy)

Reply:
> Couldn't we make it so Unity can ask kernel for realms information when it needs it?

This is probably going to be used by an UI that the user open... fetching in that moment will probably take time and the UI will be refreshed while the user has it open... I think it will look kind of ugly... but I can use that approach you suggested and if it look too bad we can change it back later

Reply:
Yeah, my bad... clearly my first intention was doing that function async... that's why it has an `await` there, I don't know what happen in the middle and I forgot 😂

Reply:
No need. If the information doesn't need to be up to date, then I think I'm OK with the regular update.

Reply:
Yeah, I think a 2 minutes interval is OK, we don't need to have the "current" precise info for now